Implementation of improvements to the original code

mrfelpa opened the pull request:
Implementation of a neural network model and the model is trained in real time using the public key as input data and the corresponding prime factors as target output

Commit:
Implementation of improvements to the original code

Implementation of a neural network model and the model is trained in real time using the public key as input data and the corresponding prime factors as target output

eddieoz commented:
Thank you for the nn code to the find primes! 

It would be great if you could submit it as a separate file, like "find_primes-nn.py", and include yourself as the contributor in the header. 

The issue is that I typically use this code to demonstrate the mathematical foundations of cryptography without using any libraries. If the neural network code is in a separate file, I can show both the mathematical approach and how to use neural networks to solve the same problem. This would be an impressive demonstration for the audience.

I have a question - what Python version are you using? I wasn't able to run the code with Python 3.12 or 3.11. It's likely an issue with the TensorFlow version being incompatible with those Python versions.

ghost commented:
Hello @eddieoz, how are you? I follow morning crypto every morning

Thanks for the suggestion to implement it separately. I'll do it then as soon as I free up some time here. Yes, the version used was Python 3.6. However, the idea is to improve the code to be compatible with other functions

See you later and have a good day

Commit:
CreateNewCode

The tool uses a neural network model to predict the prime factors of the RSA public key. If the prediction fails, the tool resorts to a classic prime search algorithm to guarantee the result

mrfelpa reviewed (commented):
- Auxiliary Mathematical Functions. 
- A sequential Keras model with the following layers:
Input layer: 1 neuron
Hidden layers: 64, 32 and 16 neurons with ReLU activation
Output layer: 2 neurons with linear activation

- The model is compiled using the Adam optimizer and the MSE (Mean Squared Error) loss function.

![I A_RSA](https://github.com/user-attachments/assets/d1e7eb93-555c-4924-bfd6-76002104547d)

Commit:
UpdateREADME.md

Description of future improvements added

eddieoz commented:
That's great!! Thank you very much :)

Are you interested in presenting your contribution and explain how it works during the MC? What you think? 
I believe the community will like a lot to know about this improvement!

ghost commented:
Hi Eddie, how are you?

Sorry for the delay in responding, I'm in a rush finishing my postgraduate work and preparing for my master's degree lol. Add that up and there's almost no time left over for anything.

I found you there at keybase, I'll call you to exchange ideas.
